### PR TITLE
Use osfamily to select the correct params values

### PR DESCRIPTION
**Pull request by robinbowes**
- specifically, so this works on CentOS

### CODE REVIEW

**Comment by derdanne** [comment]
Hey Rainbowes, 

thanks, i did not thought about that!

Could you please also do your changes in the spec tests, so that Travis builds will pass?

**Comment by derdanne** [comment]
@robinbowes 

ping

**Comment by robinbowes** [comment]
Done.

**Comment by derdanne** [comment]
Hi @robinbowes, thanks for your work, looks pretty good! 

Sadly i have another little change request: 

Recently i changed the fail in params.pp to notices, so the module works also on other operatingsystems when all parameters are specified correctly. Could you please change this in your pull request?

**Comment by robinbowes** [comment]
If you're going to do that then you should also set all the variables to "undef" (so your module works with strict variables turned on). Want me to add that too?

**Comment by derdanne** [comment]
Sounds like a good idea :) Would be great if you could do that!

**Comment by robinbowes** [comment]
OK, so I updated the params class to set parameters for unknown OSes to undef. I also noticed that changes were required on EL7 vs. EL6 (and presumably earlier) so I modified the case statement to switch on combined osfamily and major OS release version and added two separate cases for RedHat-7 and all other RedHat.

However, there's one test fail that I can't squish. I'm leaving it for now and will re-visit later, but feel free to take a look and see if you can see what's causing the failure.

File: spec/defines/server_export_spec.rb [hunk 1]
```diff
@@ -3,7 +3,14 @@
 describe 'nfs::server::export', :type => 'define' do
 
   context "nvs_v4 => false" do
-    let(:facts) {{ :operatingsystem => 'ubuntu', :concat_basedir => '/dne', :osfamily => 'debian', :clientcert => 'test.host' }}
+    let(:facts) {{
+      :operatingsystem => 'Ubuntu',
+      :osfamily => 'Debian',
+      :operatingsystemmajrelease => '12.04',
+      :concat_basedir => '/dne',
+      :osfamily => 'debian',
```
Comment (derdanne) [inline spec/defines/server_export_spec.rb]: This line is causing the failure - it is overwriting 'Debian' - that lowercase 'debian' is not hitting in params.pp.

File: spec/defines/server_export_spec.rb [hunk 2]
```diff
@@ -3,7 +3,14 @@
...
+      :clientcert => 'test.host',
```
Comment (derdanne) [inline spec/defines/server_export_spec.rb]: I think we can remove this line.

**Comment by derdanne** [comment]
I commented inline, i think i found the line that is causing the failure in travis ci. 

The rest looks very good, thanks for the work again!

**Comment by robinbowes** [comment]
Aha! Thanks - I knew it would be something stupid related to the case statements not working.

I'll fix that up and push the changes later tonight.

I've got another issue relating to the /etc/exports concat resource which I'll cover in a separate ticket.

**Comment by robinbowes** [comment]
Yay! All fixed!

**Comment by derdanne** [comment]
Yay, thanks! 

... and merged :)

**Pull request merged**